.github/workflows/CI.yml

```
@@ -123,6 +123,25 @@ jobs:
           fi
         shell: bash
 
+  coverage:
+    runs-on: ubuntu-latest
+    env:
+      CARGO_TERM_COLOR: always
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Rust
+        run: rustup update stable
```

```suggestion
      - uses: actions/cache/restore@v4
        with:
          path: ${{ env.RUST_CACHE_PATH }}
          key: cargo-x86_64-unknown-linux-gnu
          restore-keys: |
            cargo-x86_64-unknown-linux-gnu
```

Note: Rust is already installed on ubuntu images.